clusterloader2/pkg/measurement/util/controlled_pods_indexer.go

```diff
@@ -40,8 +40,191 @@ type ControlledPodsIndexer struct {
+		return nil, fmt.Errorf("failed to register indexer: %w", err)
+	}
+
+	cpi := &ControlledPodsIndexer{
+		podsIndexer:           podsInformer.Informer().GetIndexer(),
+		podsSynced:            podsInformer.Informer().HasSynced,
+		rsUIDToState:          make(map[types.UID]*ReplicaSetState),
+		deploymentUIDToRsUIDs: make(map[types.UID]UIDSet),
+	}
+
+	podsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+				rsUID, kind := cpi.getControllerInfo(obj)
+				if rsUID == "" || kind != "ReplicaSet" {
+					return
+				}
+				cpi.updatePodRefCountLocked(rsUID, 1)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+
+				oldRsUID, oldKind := cpi.getControllerInfo(oldObj)
+				newRsUID, newKind := cpi.getControllerInfo(newObj)
+				if oldRsUID == newRsUID {
+					return
+				}
+				if oldKind == "ReplicaSet" {
+					cpi.updatePodRefCountLocked(oldRsUID, -1)
+				}
+				if newKind == "ReplicaSet" {
+					cpi.updatePodRefCountLocked(newRsUID, 1)
+				}
+				if oldKind == "ReplicaSet" {
+					cpi.clearRSDataIfPossibleLocked(oldRsUID)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+
+				rsUID, kind := cpi.getControllerInfo(obj)
+				if rsUID == "" || kind != "ReplicaSet" {
+					return
+				}
+				cpi.updatePodRefCountLocked(rsUID, -1)
+				cpi.clearRSDataIfPossibleLocked(rsUID)
+			},
+		},
+	)
+	rsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+
+				deploymentUID, kind := cpi.getControllerInfo(obj)
+				if deploymentUID == "" || kind != "Deployment" {
+					return
+				}
+				cpi.handleIncomingRSAddEventLocked(obj)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+
+				oldDeploymentUID, oldKind := cpi.getControllerInfo(oldObj)
+				newDeploymentUID, newKind := cpi.getControllerInfo(newObj)
+				if oldDeploymentUID == newDeploymentUID {
+					return
+				}
+				if oldKind == "Deployment" {
+					cpi.handleIncomingRSDeleteEventLocked(oldObj)
+				}
+				if newKind == "Deployment" {
+					cpi.handleIncomingRSAddEventLocked(newObj)
+				}
+				if oldKind == "Deployment" {
+					cpi.clearRSDataIfPossibleLocked(cpi.getObjUID(oldObj))
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				cpi.UIDLock.Lock()
+				defer cpi.UIDLock.Unlock()
+
+				deploymentUID, kind := cpi.getControllerInfo(obj)
+				if deploymentUID == "" || kind != "Deployment" {
+					return
+				}
+				cpi.handleIncomingRSDeleteEventLocked(obj)
+				cpi.clearRSDataIfPossibleLocked(cpi.getObjUID(obj))
+			},
+		},
+	)
+
+	return cpi, nil
+}
+
+func (p *ControlledPodsIndexer) getControllerInfo(obj interface{}) (types.UID, string) {
```

Comment:
Nit: it doesn't need to be a member of ControlledPodsIndexer as it never accesses any field + seems to be reusable outside of the ControlledPodsIndexer

Please migrate to:

func getControllerInfo(obj interface{})

The same for getObjUID